### PR TITLE
Cover revise leftovers for comparison operations

### PR DESCRIPTION
**Pull request by nsemaev**
### Details:
 - **Removed unnecessary reference implementation overload** *(Equal, Greater, GreaterEq, Less, LessEq)*
 - **Removed backend tests** (*Equal, Greater, GreaterEq, Less, LessEq)*
 - **Removed visit_attributes() function** *(Equal)*
 - **Added NGRAPH_CHECK macros in evaluate() functions** *(Equal, Greater, Less, LessEq)*

### Tickets:
 - *63898*

### CODE REVIEW

**Review by mitruska — APPROVED** [review]

File: ngraph/test/backend/constant.in.cpp [hunk 1]
```diff
@@ -146,26 +146,6 @@ NGRAPH_TEST(${BACKEND_NAME}, tensor_constant_int64) {
     EXPECT_EQ((vector<int64_t>{0x4000000000000001, 0x4000000000000002}), read_vector<int64_t>(result));
 }
 
-NGRAPH_TEST(${BACKEND_NAME}, constant_equality_bool) {
```
Comment (ilyachur) [inline ngraph/test/backend/constant.in.cpp:149]: Why did you remove this test?

**Comment by akladiev** [comment]
This PR will be closed in 2 weeks in case of no activity.

**Comment by akladiev** [comment]
This PR was closed because it has been stalled for 2 week with no activity.
